store/multi_writer_app_store_test.go

```diff
@@ -269,3 +278,82 @@ func mockMultiWriterStore(flushInterval int64) (*MultiWriterAppStore, error) {
 func vmPrefixKey(key string) []byte {
 	return util.PrefixKey([]byte("vm"), []byte(key))
 }
+
+func TestMultiWriterSnapshots(t *testing.T) {
+	numBlocks = 25
+	blockSize = 5
+	flushInterval := int64(10)
+	prefix := []byte("test")
+	log.Setup("debug", "file://-")
+	log.Root.With("module", "dual-iavlstore")
+
+	blocks = nil
+	blocks = iavl.GenerateBlocksHashKeys(numBlocks, blockSize, append(prefix, byte(0)))
+
+	controlStore, err := mockMultiWriterStore(flushInterval)
+	var controlHashes [][]byte
+	require.NoError(t, err)
+	for _, block := range blocks {
+		require.NoError(t, block.Execute(controlStore.appStore.tree))
+		hash, _, err := controlStore.SaveVersion()
+		controlHashes = append(controlHashes, hash)
+		require.NoError(t, err)
+	}
+
+	testStore, err := mockMultiWriterStore(flushInterval)
+	require.NoError(t, err)
+
+	quit := make(chan bool)
+	var wg sync.WaitGroup
+	wg.Add(2)
+	var testHashes [][]byte
+
+	go func(s *MultiWriterAppStore) {
+		defer wg.Done()
+		for _, block := range blocks {
+			require.NoError(t, block.Execute(s.appStore.tree))
+			hash, _, err := s.SaveVersion()
+			require.NoError(t, err)
+			testHashes = append(testHashes, hash)
+			time.Sleep(5 * time.Millisecond)
+		}
+		quit <- true
+	}(testStore)
+
+	go func(s *MultiWriterAppStore) {
+		defer wg.Done()
+		for {
+			select {
+			case <-quit:
+				return
+			default:
+				{
+					snapshot := s.GetSnapshot()
+					for _, data := range snapshot.Range(prefix) {
+						temp = data.Key
+						temp = data.Value
```

Comment:
What's the point of assigning to `temp` here?

Reply:
In case we want to address `-race` issues. Until these are fixed setting these values will generate errors when `go test` is run with the `-race` option.
`temp` is global to make sure something actually happens here, and the compiler does not optimise everything away.